### PR TITLE
Do not chain set/get on WeakMaps

### PR DESCRIPTION
**Pull request by thomaswilburn**
WeakMaps do not return themselves in IE for chaining

### CODE REVIEW

File: can-dom-mutate.js [hunk 1]
```diff
@@ -29,7 +29,11 @@ function getRelatedData(node, key) {
 }
 
 function setRelatedData(node, key, targetListenersMap) {
-	var data = dataStore.get(node) || dataStore.set(node, {}).get(node);
+	var data = dataStore.get(node)
+	if (!data) {
+		dataStore.set(node, {})
+		data = {};
```
Comment (matthewp) [inline can-dom-mutate.js]: shouldn't `data` be the thing you pass into `dataStore.set`?

Comment (thomaswilburn) [reply can-dom-mutate.js]: I don't think so, it's creating the entry in the store only if it doesn't exist for the key of `node`.

Comment (thomaswilburn) [reply can-dom-mutate.js]: Ah, but if we want it to be an object reference... I see what you're saying. Will fix.

**Comment by matthewp** [comment]
Linting errors are what's blocking this.

**Comment by matthewp** [comment]
This is good to go!

**Pull request merged**